__init__.py

```diff
@@ -663,71 +686,47 @@ def _get_alarm_matches(
-
-        when, utt_no_datetime = extract_datetime(utt) or (None, utt)
-        self.log.debug("when: %s (this is in local timezone), what: %s" % (when, utt_no_datetime))
-
-        when_utc = when.astimezone(pytz.utc)
-        self.log.debug("when_utc: %s (this is in utc)" % (when_utc,))
-
-        have_time = False
-        if when_utc is None:
-            self.log.warning("Request contains no discernable date or time")
-        else:
-            # assumption 1 - if time is midnight we probably don't have a time
-            # in other words the user probably said something like 'monday the 25th'
-            # or tomorrow. 
-            # we can look for the term 'midnight' in the utterance and
-            # if present we can assume we DO have a time (midnight = 00:00:00), 
-            # otherwise we assume we don't have a time
-            user_time = when.strftime("%H:%M:%S")
-            if user_time == "00:00:00" and utt.find("midnight") == -1:
-                self.log.debug("we have no user time ===> 00:00:00")
+    # sometimes LF returns a roman numeral for
+    # a number so the 26th will return xxvi
+    def roman_to_int(self, s):
+        rom_val = {'I': 1, 'V': 5, 'X': 10, 'L': 50, 'C': 100, 'D': 500, 'M': 1000}
+        int_val = 0
+        for i in range(len(s)):
+            if i > 0 and rom_val[s[i]] > rom_val[s[i - 1]]:
+                int_val += rom_val[s[i]] - 2 * rom_val[s[i - 1]]
             else:
-                have_time = True
-
-        self.log.debug("Do we have a user time:%s" % (have_time,))
-
-        alarm_matches = []
+                int_val += rom_val[s[i]]
+        return int_val
 
-        user_dow = None   # if user said a day of the week, this is it
-
-        result = re.findall('(mon|tues|wed|thurs|fri|sat|sun)day', utt)
-        if result:
-            utt_dow = result[0] + 'day'
-            user_dow = days.index(utt_dow) if utt_dow in days else None
+    def get_advanced_matches(self, utt, have_time, when, when_utc, user_dow, alarm_list):
+        # holds exact date and time matches
+        exact_matches = []
+        dom = None
+        try:
+            dom = self.roman_to_int(utt)
+        except:
+            self.log.debug("Not a roman numeral")
 
-        self.log.debug("user_dow:%s" % (user_dow,))
+        if dom is None:
+            dom = extract_number(utt)
 
         # if the user says a day of the week (mon-sun) we 
         # keep them in a separate list of day of week matches
         dow_matches = []
 
+        # we also look for specific time matches like 9am
+        tod_matches = []
+        if when_utc:
+            tod_matches = self.get_tod_matches(when.time(), alarm_list)
+
+        # these alarms match our when_utc date only
+        date_matches = []
+
+        # these alarms match any numbers passed in like
+        # the 25th, 5 etc which we assume are dom
+        # hopefully this won't muss up ordinals
+        dom_matches = []
```

Comment:
What happens with an utterance like "cancel my alarm at 5"?

Reply:
You will get 5 most times. Its just things like the 5th, the 26th which seem to cause the issue.

Reply:
The number extractors will only extract ordinal numbers when they are passed `ordinals=True`

Reply:
Feels like a number should only be considered a `day_of_the_month` if it's an ordinal:
"set alarm for 8am on the 5th"
or if a month is included in the utterance but would need to be careful with that:
"set alarm for 8am on June 5"

^^ Does anyone actually say this? "June 5"
Maybe we should stick with ordinals only to be safe. No ones going to say "set 5th alarm for 8am" or "set alarm for 5th hour of the day" < these are the best I could come up with where the ordinal wasn't a day of the month...

Reply:
Actually I believe ordinal is used to identify an index in a list

Reply:
That's one use case. It's used to pull numbers with a suffix. `ordinals=False` will hit "3" but not "3rd"; `ordinals=True` should do the opposite.

It's imperfect, but in a case like this, I'd do two passes. Time of day will almost never be expressed as an ordinal; as Gez noted, dates almost always will.